handlers/secrets.go

```diff
@@ -75,3 +86,221 @@ func makeSecretsArray(c *client.Client, secretNames []string) ([]*swarm.SecretRe
+	return http.StatusOK, resultsJson, nil
+}
+
+func createNewSecret(c *client.Client, body []byte) (responseStatus int, responseBody []byte, err error) {
+	var secret requests.Secret
+
+	unmarshalErr := json.Unmarshal(body, &secret)
+	if unmarshalErr != nil {
+		return http.StatusBadRequest,
+			nil,
+			fmt.Errorf("error unmarshalling body to json in secretPostHandler: %s", unmarshalErr)
+	}
+
+	_, createSecretErr := c.SecretCreate(context.Background(), swarm.SecretSpec{
+		Annotations: swarm.Annotations{
+			Name: secret.Name,
+			Labels: map[string]string{
+				ownerLabel: ownerLabelValue,
+			},
+		},
+		Data: []byte(secret.Value),
+	})
+	if createSecretErr != nil {
+		return http.StatusInternalServerError, nil, fmt.Errorf(
+			"error creating secret in secretPostHandler: %s",
+			createSecretErr,
+		)
+	}
+
+	return http.StatusCreated, nil, nil
+}
+
+func updateSecret(c *client.Client, body []byte) (responseStatus int, responseBody []byte, err error) {
+	var secret requests.Secret
+
+	unmarshalErr := json.Unmarshal(body, &secret)
+	if unmarshalErr != nil {
+		return http.StatusBadRequest, nil, fmt.Errorf(
+			"error unmarshaling secret in secretPutHandler: %s",
+			unmarshalErr,
+		)
+	}
+
+	foundSecret, getSecretErr, status := getSecretWithName(c, secret.Name)
+	if getSecretErr != nil {
+		return status, nil, fmt.Errorf(
+			"cannot get secret with name: %s, which you want to remove. Error: %s",
+			secret.Name,
+			getSecretErr,
+		)
+	}
+
+	nextVersion := swarm.Version{
+		Index: foundSecret.Version.Index + 1,
+	}
+
+	updateSecretErr := c.SecretUpdate(context.Background(), foundSecret.ID, nextVersion, swarm.SecretSpec{
+		Annotations: swarm.Annotations{
+			Name: secret.Name,
+		},
+		Data: []byte(secret.Value),
+	})
+	if updateSecretErr != nil {
+		return http.StatusInternalServerError, nil, fmt.Errorf(
+			"couldn't update secret (name: %s, ID: %s) because of error: %s",
+			secret.Name,
+			foundSecret.ID,
+			getSecretErr,
+		)
+	}
+
+	return http.StatusOK, nil, nil
+}
+
+func deleteSecret(c *client.Client, body []byte) (responseStatus int, responseBody []byte, err error) {
+	var secret requests.Secret
+
+	unmarshalErr := json.Unmarshal(body, &secret)
+	if unmarshalErr != nil {
+		return http.StatusBadRequest,
+			nil,
+			fmt.Errorf("error unmarshaling secret in secretDeleteHandler: %s", unmarshalErr)
+	}
+
+	foundSecret, getSecretErr, status := getSecretWithName(c, secret.Name)
+	if getSecretErr != nil {
+		return status, nil, fmt.Errorf(
+			"cannot get secret with name: %s, which you want to remove. Error: %s",
+			secret.Name,
+			getSecretErr,
+		)
+	}
+
+	removeSecretErr := c.SecretRemove(context.Background(), foundSecret.ID)
+	if removeSecretErr != nil {
+		return http.StatusInternalServerError, nil, fmt.Errorf(
+			"error trying to remove secret (name: `%s`, ID: `%s`): %s",
+			secret.Name,
+			foundSecret.ID,
+			removeSecretErr,
+		)
+	}
+
+	return http.StatusOK, nil, nil
+}
+
+var secretHandlers = map[string]secretHandler{
```

Comment:
This won't change, so please can you put it back inside the method body?

Reply:
Done